packages/jsii/lib/compiler.ts

```diff
@@ -72,28 +69,57 @@ export class Compiler implements Emitter {
   }
 
   /**
-     * Compiles the configured program.
-     *
-     * @param files can be specified to override the standard source code location logic. Useful for example when testing "negatives".
-     */
+   * Compiles the configured program.
+   *
+   * @param files can be specified to override the standard source code location logic. Useful for example when testing "negatives".
+   */
   public async emit(...files: string[]): Promise<EmitResult | never> {
+    await this._prepareForBuild(...files);
+    return this._buildOnce();
+
+  }
+
+  /**
+   * Watches for file changes and dynamically re-compiles whenever needed.
+   */
+  public async watch(
+    onStatusChanged?: ts.WatchStatusReporter,
+    onDiagnostic = logDiagnostic,
+  ): Promise<CompilerWatch> {
+    await this._prepareForBuild();
+
+    return new CompilerWatch(
```

Comment:
Instead of wrapping the typescript API, why not just expose the TS watch object as @internal?

Reply:
I like containing the TypeScript compiler API as much as I can because it's not been very stable in the past.